deploy/stacks/lambda_api.py

```diff
@@ -223,23 +228,25 @@ def __init__(
                 )
             )
 
-        if custom_auth is not None:
-            # Create the custom authorizer lambda
-            custom_authorizer_assets = os.path.realpath(
-                os.path.join(
-                    os.path.dirname(__file__),
-                    '..',
-                    'custom_resources',
-                    'custom_authorizer',
-                )
+        # Create the custom authorizer lambda
+        custom_authorizer_assets = os.path.realpath(
+            os.path.join(
+                os.path.dirname(__file__),
+                '..',
+                'custom_resources',
+                'custom_authorizer',
             )
+        )
+        ## GET CONGITO USERL POOL ID and APP CLIENT ID
 
-            if not os.path.isdir(custom_authorizer_assets):
-                raise Exception(f'Custom Authorizer Folder not found at {custom_authorizer_assets}')
+        if not os.path.isdir(custom_authorizer_assets):
+            raise Exception(f'Custom Authorizer Folder not found at {custom_authorizer_assets}')
 
+        if custom_auth:
```

Comment:
Just an idea for the future:
What makes cognito a special authorizer? Perhaps we should treat it as a generic OAuth2.0 provider and avoid duplicating logic. For convenience customers can chose to enable cognito and then we can automatically populate the env variables, otherwise they have to provide them,

Reply:
agreed we should just document it as a generic OAuth2.0 irrespective of whatever IdP